src/GCDNotifier.cs

```diff
@@ -0,0 +1,308 @@
+using GCDTracker.UI;
+using System.Collections.Generic;
+using Dalamud.Interface.Animation;
+using Dalamud.Interface.Animation.EasingFunctions;
+using System.Numerics;
+using System.Linq;
+using System;
+
+namespace GCDTracker {
+
+    public enum EventType {
+        FlyOutAlert,
+        BarColorPulse,
+        BarWidthPulse,
+        BarHeightPulse,
+        BarBackground,
+        WheelPulse,
+        FloatingTriangle,
+        None
+    }
+
+    public enum EventCause {
+        Slidecast,
+        Queuelock,
+        Clipped,
+        ABC,
+        None
+    }
+
+    public class Alert
+    {
+        public EventType Type { get; set; }
+        public EventCause Reason { get; set; }
+        public float LocationX { get; set; }
+        public float LocationY { get; set; }
+        public float LastClipDelta { get; set; }
+
+        public Alert(EventType type, EventCause reason, float lastClipDelta, float locationX, float locationY)
+        {
+            Type = type;
+            Reason = reason;
+            LocationX = locationX;
+            LocationY = locationY;
+            LastClipDelta = lastClipDelta;
+        }
+    }
+
+    public class AlertManager
+    {
+        private static AlertManager instance;
+        private readonly Queue<Alert> alertQueue = new Queue<Alert>();
+        private AlertManager() { }
+
+        public static AlertManager Instance => instance ??= new AlertManager();
+
+        public int AlertCount => alertQueue.Count;
+
+        public void AddAlert(EventType type, EventCause reason, float locationX = 0f, float locationY = 0f, float lastClipDelta = 0f) {
```

Comment:
I think the AlertManager should be the one to block adding duplicates.

Reply:
I've done this by toggling the alert that is already in memory to be active instead of adding a new alert.  There can never be a duplicate in this case.